python/langsmith/run_trees.py

```diff
@@ -400,12 +413,66 @@ def _get_dicts_safe(self):
             self_dict["outputs"] = self.outputs.copy()
         return self_dict
 
+    def _remap_for_project(
+        self, project_name: str, updates: Optional[dict] = None
+    ) -> dict:
+        """Rewrites ids/dotted_order for a given project with optional updates."""
+        run_dict = self._get_dicts_safe()
+        old_id = run_dict["id"]
+        new_id = uuid5(NAMESPACE_DNS, f"{old_id}:{project_name}")
+        # trace id
+        old_trace = run_dict.get("trace_id")
+        if old_trace:
+            new_trace = uuid5(NAMESPACE_DNS, f"{old_trace}:{project_name}")
+        else:
+            new_trace = None
+        # parent id
+        parent = run_dict.get("parent_run_id")
+        if parent:
+            new_parent = uuid5(NAMESPACE_DNS, f"{parent}:{project_name}")
+        else:
+            new_parent = None
+        # dotted order
+        if run_dict.get("dotted_order"):
+            segs = _parse_dotted_order(run_dict["dotted_order"])
+            rebuilt = []
+            for ts, seg_id in segs[:-1]:
+                repl = uuid5(NAMESPACE_DNS, f"{seg_id}:{project_name}")
+                rebuilt.append(ts.strftime("%Y%m%dT%H%M%S%fZ") + str(repl))
+            ts_last, _ = segs[-1]
+            rebuilt.append(ts_last.strftime("%Y%m%dT%H%M%S%fZ") + str(new_id))
+            dotted = ".".join(rebuilt)
+        else:
+            dotted = None
+        dup = utils.deepish_copy(run_dict)
+        dup.update(
+            {
+                "id": new_id,
+                "trace_id": new_trace,
+                "parent_run_id": new_parent,
+                "dotted_order": dotted,
+                "session_name": project_name,
+            }
+        )
+        if updates:
+            dup.update(updates)
```

Comment:
Why not just do this once in `post`?

This will not apply if you're posting a replica to the same project as `project_name`